### PR TITLE
Add FileProfile for addressing files (fixes #287)

### LINKED ISSUE #287: FileProfile for addressing application data files required
Issue #105 for Splitting up the LOADfile is requesting for:
>
> - If application data is stored in one or several files, name and location of these files shall also be seen as part of the configuration, which is to be stored in the CONFIGfile.
>

- Adding the following entry into the [ProfileList](https://github.com/openBackhaul/ApplicationPattern/blob/develop/ApplicationPattern+profiles.yaml):
```
  - profile-name: 'FileProfile'
    uuid: '^[a-z]{2,6}-([0-9]+)-([0-9]+)-([0-9]+)-file-p-[0-9]{3}$'
    capability:
      file-identifier: string
      file-description: string
    configuration:
      file-path: string
      user-name: string
      password: string
      operation: string
```

- Adding the following entry into the [ProfileInstanceList](https://github.com/openBackhaul/ApplicationPattern/blob/develop/ApplicationPattern+profileInstances.yaml):
```
  - profile-name: 'FileProfile'
    uuid: '*-file-p-???' ## TODO: Replace "*" by ApplicationID and ReleaseNumber and ??? by SequenceNumber
    capability:
      file-identifier: ## TODO: Add invariant identifier of the referenced file
      file-description: ## TODO: Add a description of the content of the referenced file
    configuration:
      file-path: ## TODO: Add file name, like e.g. '../application-data/application-data.json'
      user-name: ## TODO: If file protected, add user-name
      password: ## TODO: If file protected, add password
      operation: ## TODO: Choose from read/write/off

  ## TODO: Potentially add further instances of FileProfile
```

- OAS
  - Adding the following entry into the OAM Layer - Individual Part: /core-model-1-4:control-construct: -> control-construct/profile-collection/profile/  anyOf:
  ```
	- description: 'file profile'
	  type: object
	  required:
		- uuid
		- profile-name
		- file-profile-1-0:file-profile-pac
	  properties:
		uuid:
		  type: string
		profile-name:
		  type: string
		  enum:
			- 'action-profile-1-0:PROFILE_NAME_TYPE_ACTION_PROFILE'
			- 'file-profile-1-0:PROFILE_NAME_TYPE_FILE_PROFILE'
			- 'integer-profile-1-0:PROFILE_NAME_TYPE_INTEGER_PROFILE'
			- 'string-profile-1-0:PROFILE_NAME_TYPE_STRING_PROFILE'
		file-profile-1-0:file-profile-pac:
		  type: object
		  required:
			- file-profile-capability
			- file-profile-configuration
		  properties:
			file-profile-capability:
			  type: object
			  required:
				- file-identifier
				- file-description
			  properties:
				file-identifier:
				  type: string
				file-description:
				  type: string
			file-profile-configuration:
			  type: object
			  required:
				- file-path
			  properties:
				file-path:
				  type: string
				user-name:
				  type: string
				password:
				  type: string
				operation:
				  type: string
				  enum:
					- 'file-profile-1-0:OPERATION_TYPE_READ_ONLY'
					- 'file-profile-1-0:OPERATION_TYPE_READ_WRITE'
					- 'file-profile-1-0:OPERATION_TYPE_OFF'
					- 'file-profile-1-0:OPERATION_TYPE_NOT_YET_DEFINED'
  ```

  - Adding the following entry into the OAM Layer - Individual Part: /core-model-1-4:control-construct/profile-collection/profile={uuid}: -> core-model-1-4:profile oneOf:
  ```
	  - description: 'file profile'
		type: object
		required:
		  - core-model-1-4:uuid
		  - core-model-1-4:profile-name
		  - file-profile-1-0:file-profile-1-0:file-profile-pac
		properties:
		  core-model-1-4:uuid:
			type: string
		  core-model-1-4:profile-name:
			type: string
			enum:
		  	  - 'action-profile-1-0:PROFILE_NAME_TYPE_ACTION_PROFILE'
			  - 'file-profile-1-0:PROFILE_NAME_TYPE_FILE_PROFILE'
			  - 'integer-profile-1-0:PROFILE_NAME_TYPE_INTEGER_PROFILE'
			  - 'string-profile-1-0:PROFILE_NAME_TYPE_STRING_PROFILE'
		  file-profile-1-0:file-profile-pac:
			type: object
			required:
			  - file-profile-capability
			  - file-profile-configuration
			properties:
			  file-profile-capability:
				type: object
				required:
				  - file-identifier
				  - file-description
				properties:
				  file-identifier:
					type: string
				  file-description:
					type: string
			  file-profile-configuration:
				type: object
				required:
				  - file-path
				properties:
				  file-path:
					type: string
				  user-name:
					type: string
				  password:
					type: string
				  operation:
					type: string
					enum:
					  - 'file-profile-1-0:OPERATION_TYPE_READ_ONLY'
					  - 'file-profile-1-0:OPERATION_TYPE_READ_WRITE'
					  - 'file-profile-1-0:OPERATION_TYPE_OFF'
					  - 'file-profile-1-0:OPERATION_TYPE_NOT_YET_DEFINED'
  ```

  - Adding the following paths to the OAM Layer - Individual Part:
    -   /core-model-1-4:control-construct/profile-collection/profile={uuid}/file-profile-1-0:file-profile-pac/file-profile-capability/file-identifier:
    - /core-model-1-4:control-construct/profile-collection/profile={uuid}/file-profile-1-0:file-profile-pac/file-profile-capability/file-description:
    - /core-model-1-4:control-construct/profile-collection/profile={uuid}/file-profile-1-0:file-profile-pac/file-profile-configuration/file-path:
    - /core-model-1-4:control-construct/profile-collection/profile={uuid}/file-profile-1-0:file-profile-pac/file-profile-configuration/user-name:
    - /core-model-1-4:control-construct/profile-collection/profile={uuid}/file-profile-1-0:file-profile-pac/file-profile-configuration/password:
    - /core-model-1-4:control-construct/profile-collection/profile={uuid}/file-profile-1-0:file-profile-pac/file-profile-configuration/operation:
 
- No change to the CONFIGfile, because applying a APPDATAfile is individual to the application.

### PR DESCRIPTION
**Pull request by openBackhaul**

### CODE REVIEW

File: doc/SpecifyingApplications/CreatingProfileInstanceList/CreatingProfileInstanceList.md [hunk 1]
```diff
@@ -30,9 +30,8 @@ This is a step by step cookbook for creating the ProfileInstanceList.
 * The UUID shall follow the [Structure of UUIDs](../../ElementsApplicationPattern/Names/StructureOfUuids/StructureOfUuids.md).  
 * The LayerID shall be set on "action".  
 * The ObjectType shall be set on "p" (Profile).  
-* The ApiSegment shall be set on "1" (management, which is individual to this application).  
-* The ApplicationNumber shall be set on "0" (the application itself).  
-* The SequenceNumber shall start at "00" and must be unique within the scope of instances of Profiles with the same LayerID.  
+* The ApiSegment shall be set on "mi" (management, which is individual to this application).
```
Comment (PrathibaJee) [inline doc/SpecifyingApplications/CreatingProfileInstanceList/CreatingProfileInstanceList.md:33]: As per the Structure of UUIDs diagram(uuid-structure.png) and the concept under the ApiSegment , “UUIDs of Profile, ForwardingDomain and Link don't identify an ApiSegment.”
In action profile , 
The ApiSegment shall be set on "mi" (management, which is individual to this application). 

This statement is contradicting with the concept in the Structure of uuid

Either the uuid-structure needs to be corrected 
Or
This line should be deleted

File: doc/SpecifyingApplications/CreatingProfileInstanceList/CreatingProfileInstanceList.md [hunk 2]
```diff
@@ -60,6 +59,49 @@ This is a step by step cookbook for creating the ProfileInstanceList.
...
+* Finally, just delete the form that waits for completion.  
+
+**ProfileName**  
+* The ProfileName shall be set on "FileProfile".  
+
+**UUID**  
+* The UUID shall follow the [Structure of UUIDs](../../ElementsApplicationPattern/Names/StructureOfUuids/StructureOfUuids.md).  
+* The LayerID shall be set on "file".  
+* The ObjectType shall be set on "p" (Profile).  
+* The ApiSegment shall be set on "im" (management, which is individual to this application).
```
Comment (PrathibaJee) [inline doc/SpecifyingApplications/CreatingProfileInstanceList/CreatingProfileInstanceList.md:75]: Same as mentioned in the action profile , 

This statement is contradicting with the concept in the Structure of uuid

Either the uuid-structure needs to be corrected 
Or
This line should be deleted

File: doc/SpecifyingApplications/CreatingProfileInstanceList/CreatingProfileInstanceList.md [hunk 3]
```diff
@@ -73,9 +115,8 @@ This is a step by step cookbook for creating the ProfileInstanceList.
 * The UUID shall follow the [Structure of UUIDs](../../ElementsApplicationPattern/Names/StructureOfUuids/StructureOfUuids.md).  
 * The LayerID shall be set on "integer".  
 * The ObjectType shall be set on "p" (Profile).  
-* The ApiSegment shall be set on "1" (management, which is individual to this application).  
-* The ApplicationNumber shall be set on "0" (the application itself).  
-* The SequenceNumber shall start at "00" and must be unique within the scope of instances of Profiles with the same LayerID.  
+* The ApiSegment shall be set on "mi" (management, which is individual to this application).
```
Comment (PrathibaJee) [inline doc/SpecifyingApplications/CreatingProfileInstanceList/CreatingProfileInstanceList.md:118]: Same as mentioned in the action profile , 

This statement is contradicting with the concept in the Structure of uuid

Either the uuid-structure needs to be corrected 
Or
This line should be deleted

File: doc/SpecifyingApplications/CreatingProfileInstanceList/CreatingProfileInstanceList.md [hunk 4]
```diff
@@ -109,9 +150,8 @@ This is a step by step cookbook for creating the ProfileInstanceList.
 * The UUID shall follow the [Structure of UUIDs](../../ElementsApplicationPattern/Names/StructureOfUuids/StructureOfUuids.md).  
 * The LayerID shall be set on "string".  
 * The ObjectType shall be set on "p" (Profile).  
-* The ApiSegment shall be set on "1" (management, which is individual to this application).  
-* The ApplicationNumber shall be set on "0" (the application itself).  
-* The SequenceNumber shall start at "00" and must be unique within the scope of instances of Profiles with the same LayerID.  
+* The ApiSegment shall be set on "mi" (management, which is individual to this application).
```
Comment (PrathibaJee) [inline doc/SpecifyingApplications/CreatingProfileInstanceList/CreatingProfileInstanceList.md:153]: Same as mentioned in the action profile , 

This statement is contradicting with the concept in the Structure of uuid

Either the uuid-structure needs to be corrected 
Or
This line should be deleted

**Review by PrathibaJee — CHANGES_REQUESTED** [review]

File: ApplicationPattern+profileInstances.yaml [hunk 5]
```diff
@@ -42,6 +42,20 @@ profile-instances:
 
   ## TODO: Potentially add further instances of ActionProfile
 
+	  
+  - profile-name: 'FileProfile'
+    uuid: '*-file-p-???' ## TODO: Replace "*" by ApplicationID and ReleaseNumber and ??? by SequenceNumber
+    capability:
+      file-identifier: ## TODO: Add invariant identifier of the referenced file
+	    file-description: ## TODO: Add a description of the content of the referenced file
```
[the remark this answers is not in the document]
Comment (openBackhaul) [reply ApplicationPattern+profileInstances.yaml:50]: yes, but isn't it?
In my sources it looks fine.

File: ApplicationPattern+profileInstances.yaml [hunk 6]
```diff
@@ -42,6 +42,20 @@ profile-instances:
...
+    configuration:
+      file-path: ## TODO: Add file name, like e.g. '../application-data/application-data.json'
+	    user-name: ## TODO: If file protected, add user-name
```
[the remark this answers is not in the document]
Comment (openBackhaul) [reply ApplicationPattern+profileInstances.yaml:53]: yes, but isn't it?
In my sources it looks fine.

File: ApplicationPattern+profiles.yaml [hunk 7]
```diff
@@ -12,6 +12,17 @@ profiles:
     configuration:
       request: string
 
+  - profile-name: 'FileProfile'
+    uuid: '^[a-z]{2,6}-([0-9]+)-([0-9]+)-([0-9]+)-file-p-[0-9]{3}$'
+    capability:
+      file-identifier: string
+	    file-description: string
```
[the remark this answers is not in the document]
Comment (openBackhaul) [reply ApplicationPattern+profiles.yaml:19]: yes, but isn't it.
In my sources it looks all fine.

File: ApplicationPattern+profiles.yaml [hunk 8]
```diff
@@ -12,6 +12,17 @@ profiles:
...
+    configuration:
+      file-path: string
+	    user-name: string
```
[the remark this answers is not in the document]
Comment (openBackhaul) [reply ApplicationPattern+profiles.yaml:22]: yes, but isn't it.
In my sources it looks all fine.

File: doc/SpecifyingApplications/CreatingProfileInstanceList/CreatingProfileInstanceList.md (hunk 1 above)
Comment (openBackhaul) [reply doc/SpecifyingApplications/CreatingProfileInstanceList/CreatingProfileInstanceList.md:33]: Deleted this line.

File: doc/SpecifyingApplications/CreatingProfileInstanceList/CreatingProfileInstanceList.md (hunk 2 above)
Comment (openBackhaul) [reply doc/SpecifyingApplications/CreatingProfileInstanceList/CreatingProfileInstanceList.md:75]: Deleted this line.

File: doc/SpecifyingApplications/CreatingProfileInstanceList/CreatingProfileInstanceList.md (hunk 3 above)
Comment (openBackhaul) [reply doc/SpecifyingApplications/CreatingProfileInstanceList/CreatingProfileInstanceList.md:118]: Deleted this line.

File: doc/SpecifyingApplications/CreatingProfileInstanceList/CreatingProfileInstanceList.md (hunk 4 above)
Comment (openBackhaul) [reply doc/SpecifyingApplications/CreatingProfileInstanceList/CreatingProfileInstanceList.md:153]: Deleted this line

**Pull request closed without merging**